src/test/librbd/fsx.cc

```diff
@@ -2069,6 +2069,13 @@ docompareandwrite(unsigned offset, unsigned size)
         if (o_direct)
                 size -= size % writebdy;
 
+        if (skip_partial_discard) {
+                if (!quiet && testcalls > simulatedopcount && !o_direct)
```

Comment:
I suppose `!o_direct` check is not needed here. I assume it is `if (size == 0)` specific because we may truncate size in `o_direct` case.

Probably it is a good idea to move this `if (skip_partial_discard)` block above `if (o_direct)` block or below `if (size == 0)` block as they look related.